### PR TITLE
Improve authentication - Slack only auth

### LINKED ISSUE #38: Improve authentication process
Some suggestions received:

~~- [ ] Password double-typing to improve accessibility.~~
~~- [ ] Fix duplicate alerts for account/profile verification.~~
~~- [ ] Show proper errors in login/registration forms~~
~~- [ ] Account creation form: Change the label of the button from "Submit" to "Create account"~~
~~- [ ] Use red alerts only if an error occurred, use amber or blue otherwise~~
~~- [ ] "Already an account?" -> "Already have an account?"~~ 
- [ ] Allow logins with slack. Since we are already using it for account verification it would be a good idea to let users log in directly with Slack.

### PR DESCRIPTION
**Pull request by amenat**
Remove local registration / login
Remove "verified" flags from users.
Replace login link with "Sign in with slack" button.
Remove 55 dependencies! (bcrypt and passport-local).
Passport can be fully removed in future if required, replacing it with
pure JS Slack OAuth authentication.

DB reset is suggested as user database is completely changed.

This will close #38

### CODE REVIEW

File: config/auth.js [hunk 1]
```diff
@@ -107,13 +30,30 @@ module.exports = function (passport) {
...
+							"INSERT INTO users (username, email, slackuid) \
+                                values (?, ?, ?)",
+							[profile.displayName, profile.user.email, profile.id],
+							function (err, result) {
+								if (err) {
+									done(null, false);
+									req.flash("error", "Slack login failed");
+								}
+								// return user id for deserializing.
+								done(err, { id: result.insertId });
```
Comment (arjunmuralidharan) [inline config/auth.js:54]: My installation breaks here as it seems that "result" is undefined. The script to add a dummy user also fails with "Field 'password' doesn't have a default value"

File: db/add_dummy_users.sql [hunk 2]
```diff
@@ -1,4 +1,4 @@
-INSERT INTO 
-	users(username, email, password, verified, slackuid)
+INSERT INTO
+	users(username, email, slackuid)
 VALUES
-	('userName', 'user@email.com', 'somepw1', 1, 'userslackid');
\ No newline at end of file
+	('userName', 'user@email.com', 'userslackid');
```
Comment (arjunmuralidharan) [inline db/add_dummy_users.sql:4]: This fails with error "Field 'password' doesn't have a default value"

**Review by arjunmuralidharan — CHANGES_REQUESTED** [review]
I can't complete the Slack auth. It looks like the query to retrieve users breaks if the user database is empty. I also cannot create a dummy user using the script.

File: db/add_dummy_users.sql (hunk 2 above)
Comment (amenat) [reply db/add_dummy_users.sql:4]: You need to do DB reset. Looks like you're using old database. new schema doesn't have password field.

**Comment by BlairCurrey** [comment]
Adding commentary from slack:

Some points:
- Users can now no longer update their username, it will be same as slack "display name" (not Full name) We can add feature to re-fetch new name if users want to do it, but it will always be linked with their slack display name.
- you probably already know the session problem. Until that issue is fixed, server reload will log out everyone. I plan to push PR for that today or tomorrow max.

File: views/profile.html [hunk 3]
```diff
@@ -6,43 +6,25 @@
...
+			<% if (warningMessage.length > 0) { %>
+			<div class="alert alert-warning"><%= warningMessage %></div>
 			<% } %>
-
 			<!-- Update Account -->
-			<form action="/profile?_method=PUT" method="post" class="">
+			<form>
 				<div class="form-group">
 					<label for="email">Email</label>
 					<input type="email" class="form-control" name="email" value="<%= user.email %>" disabled />
```
Comment (BlairCurrey) [inline views/profile.html:20]: Are we going to use the email field anymore? It struck me as odd when I logged in via slack and the profile page showed a disabled, blank email. I think it should be removed.

File: config/auth.js (hunk 1 above)
Comment (BlairCurrey) [reply config/auth.js:54]: I have not run into any issues so far with this after rebuilding the db and changing env variables. Logged in and performed basic tasks successfully.

File: db/add_dummy_users.sql (hunk 2 above)
Comment (arjunmuralidharan) [reply db/add_dummy_users.sql:4]: Yes - had updated the db but not my .env file...

File: config/auth.js (hunk 1 above)
Comment (arjunmuralidharan) [reply config/auth.js:54]: Resolved after updating my env - didn't realise the DB is renamed.

File: views/profile.html (hunk 3 above)
Comment (arjunmuralidharan) [reply views/profile.html:20]: Agree

**Comment by arjunmuralidharan** [comment]
A general thought now is that we can do away with the /profile page. We could just add a filter option to the /reviews page ("Only my reviews") and stick an edit button into those reviews that belong to the current user.

**Comment by BlairCurrey** [comment]
> A general thought now is that we can do away with the /profile page. We could just add a filter option to the /reviews page ("Only my reviews") and stick an edit button into those reviews that belong to the current user.

@arjunmuralidharan I would be in favor of this but we might add additional functionality in the future where it would be important. Personally I like the condensed list of reviews in the profile page (as opposed to showing all the cards) but I take your point - it isn't really necessary anymore. Additionally, a filter to only show your reviews is a good feature regardless.

I don't see the harm in getting rid of the profile view right now, but if we do, I think we should save the routes and template because there is a good chance we will want to bring it back.

File: views/profile.html (hunk 3 above)
Comment (BlairCurrey) [reply views/profile.html:20]: this is a bug. adding "identity.email" to `scope` list in `auth.js` resolves.

**Pull request merged**